### PR TITLE
fix(breadbox): Fix categorical datasets validation

### PR DESCRIPTION
**Pull request by jessica-cheng**
### Fixes:

- Validation for `allowed_values` field incorrect and broken since update to pydantic 2
- Add missing tests for dataset uploads of categorical datasets. Previous categorical dataset tests were for the old POST dataset endpoint
- Clean up some methods

### CODE REVIEW

File: breadbox/breadbox/schemas/dataset.py [hunk 1]
```diff
@@ -192,6 +178,27 @@ def check_valid_value_type_with_allowed_values(self):
...
 
+    @field_validator("allowed_values", mode="after")
+    @classmethod
+    def check_valid_allowed_values(cls, v: Optional[List[str]]):
+        """
+        Checks there are no empty strings and no repeated allowed values. Values in allowed values list are not case-sensitive.
+        """
+        if v is None:
+            return v
+        # Decision to make allowed values not case-sensitive in case user error in accidental repeats
```
Comment (jessica-cheng) [inline breadbox/breadbox/schemas/dataset.py:189]: This is carried over from our past decisions but I've written the comment to make it more visible. Example of `"true"` and `True` should be considered the same value

File: breadbox/breadbox/schemas/dataset.py [hunk 2]
```diff
@@ -192,6 +178,27 @@ def check_valid_value_type_with_allowed_values(self):
...
+        allowed_values_list_lower = [str(x).lower() for x in v]
+        allowed_values_set = set(allowed_values_list_lower)
+        print(allowed_values_set, allowed_values_list_lower)
```
Comment (pgm) [inline breadbox/breadbox/schemas/dataset.py:192]: probably should pull out the print stmt before merging in

File: breadbox/breadbox/schemas/dataset.py [hunk 3]
```diff
@@ -192,6 +178,27 @@ def check_valid_value_type_with_allowed_values(self):
...
+        if len(allowed_values_set) != len(v):
+            raise UserError(
+                msg="Make sure there are no repeats in allowed_values. Values are not considered case-sensitive",
+            )
+        for val in allowed_values_set:
+            if val == "":
+                raise UserError("Empty strings are not allowed!")
```
Comment (pgm) [inline breadbox/breadbox/schemas/dataset.py:199]: Reading this error, I worry that it doesn't provide enough context. If I saw it when doing an upload, I might not realize that this was an error related to allowed_values.

How about the following?
```
raise UserError("Empty strings are not allowed in allowed_values!")
```

File: frontend/packages/@depmap/dataset-manager/src/components/DatasetForm.tsx [hunk 4]
```diff
@@ -3,7 +3,7 @@ import * as React from "react";
 import { MatrixDatasetForm } from "./MatrixDatasetForm";
 import { TableDatasetForm } from "./TableDatasetForm";
 import { FormGroup, Radio } from "react-bootstrap";
-import { useState, useEffect, useMemo } from "react";
+import { useState, useEffect, useMemo, useCallback } from "react";
```
Comment (pgm) [inline frontend/packages/@depmap/dataset-manager/src/components/DatasetForm.tsx:6]: I'm guessing these are changes stemming from the branch being branched off of a different existing branch, correct?

**Review by pgm — APPROVED** [review]
Looks fine. Made some minor remarks and ignored all the ts/tsx changes because I'm assuming they're unrelated?

File: frontend/packages/@depmap/dataset-manager/src/components/DatasetForm.tsx (hunk 4 above)
Comment (jessica-cheng) [reply frontend/packages/@depmap/dataset-manager/src/components/DatasetForm.tsx:6]: Yes I think I merged in a branch from [this PR](https://github.com/broadinstitute/depmap-portal/pull/142) to make testing on the UI side easier. I see it's approved now yay!

File: breadbox/breadbox/schemas/dataset.py (hunk 3 above)
Comment (jessica-cheng) [reply breadbox/breadbox/schemas/dataset.py:199]: Good point!

**Pull request merged**